### PR TITLE
Refactor/neon/settings

### PR DESCRIPTION
**Pull request by Leptopoda**
handles the design part of #849
Changes to existing platforms:
- the visual density of the Settings will be higher on Linux than before
- `SelecOption`s will now be handled in popup dialogs as it is common on m3
- SettingsCategories will no longer be separated by a divider to match the m3 design 

The m3 docs themselves do not mention the modifications I made but I got inspired by the system settings on stock android

### CODE REVIEW

**Comment by Leptopoda** [comment]
I think that the entire idea of a separate `SettingsListTile` is bad.
Can we just remove it and provide it and provide normal widgets and helpers to style custom ones?

**Review by Leptopoda — COMMENTED** [review]
https://github.com/nextcloud/neon/pull/1030#discussion_r1372870104

So we don't forget those refactors

File: .cspell/dart_flutter.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 autofocus
+Cupertino
```
Comment (provokateurin) [inline .cspell/dart_flutter.txt:2]: All words in the dictionaries are lowercase

File: packages/neon/neon/lib/src/settings/widgets/settings_category.dart [hunk 2]
```diff
@@ -1,41 +1,78 @@
...
-  final List<SettingsTile> tiles;
+  final List<Widget> tiles;
+  final Widget? footer;
+  final bool hasLeading;
 
   @override
   Widget build(final BuildContext context) {
+    final theme = Theme.of(context);
+
+    switch (theme.platform) {
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/settings/widgets/settings_category.dart:24]: Should use isCupertino()

**Review by provokateurin — CHANGES_REQUESTED** [review]
Select options are a bit broken because the currently selected value is not shown in the tile. Initially I thought you forgot to add it (I checked how it works in the settings on my phone) but it looks like you added it and it's just not working.

Is there an easy way to test the cupertino layout?

**Review by provokateurin — CHANGES_REQUESTED** [review]
packages/app/pubspec.lock is not updated

**Comment by Leptopoda** [comment]
> Is there an easy way to test the cupertino layout?

Just specify the platform in the theme. There are still bigger issues with the scaffold background color mismatching but as iOS is still a WIP I will address them separately.
```dart
    return ThemeData(
        platform: TargetPlatform.iOS,
        [...]
      );
```

**Comment by Leptopoda** [comment]
Also because of some license issues with the apple font you will not be able to see the `+` icon in the add account tile unless you run it on iOS itself.

**Comment by Leptopoda** [comment]
Also because of some license issues with the apple font you will not be able to see the `+` icon in the add account tile unless you run it on iOS itself.

**Comment by provokateurin** [comment]
> Also because of some license issues with the apple font you will not be able to see the + icon in the add account tile unless you run it on iOS itself.

lol what. fucking apple...

**Review by provokateurin — APPROVED** [review]

**Pull request merged**